services/src/test/scala/cromwell/services/metadata/impl/WriteMetadataActorSpec.scala

```diff
@@ -25,15 +26,21 @@ class WriteMetadataActorSpec extends ServicesSpec("Metadata") {
 
     "Have one event and be waiting after one event is sent" in {
       actor ! Action
-      assert(actor.stateName == WaitingToWrite)
-      assert(actor.stateData == HasEvents(NonEmptyVector.fromVectorUnsafe(Action.events.toVector)))
+      eventually {
+        assert(actor.stateName == WaitingToWrite)
+        assert(actor.stateData == HasEvents(NonEmptyVector.fromVectorUnsafe(Action.events.toVector)))
+      }
     }
 
     "Have one event after batch size + 1 is reached" in {
       1 to 10 foreach { _ => actor ! Action }
-      assert(actor.stateName == WritingToDb)
+      eventually {
+        assert(actor.stateName == WritingToDb)
+      }
       actor ! Action
-      assert(actor.stateData == HasEvents(NonEmptyVector.fromVectorUnsafe(Action.events.toVector)))
+      eventually {
+        assert(actor.stateData == HasEvents(NonEmptyVector.fromVectorUnsafe(Action.events.toVector)))
```

Comment:
Is there a reason you're using runtime asserts rather than the scalacheck `should be`?